### PR TITLE
Improve std::path docs

### LINKED ISSUE #29368: API Docs: path
Part of https://github.com/rust-lang/rust/issues/29329

http://doc.rust-lang.org/std/path/

Here's what's needed to close out this issue:

- [ ] [`Components`](https://doc.rust-lang.org/std/path/struct.Components.html) should use the standard iterator boilerplate.
- [ ] [`Display`](https://doc.rust-lang.org/std/path/struct.Display.html) needs some more explanation and a link to the `display` method.
- [ ] [`Iter`](https://doc.rust-lang.org/std/path/struct.Iter.html) should use the standard iterator boilerplate.
- [ ] [`PrefixComponent`](https://doc.rust-lang.org/std/path/struct.PrefixComponent.html) needs general expansion and examples
- [ ] [`StripPrefixError`](https://doc.rust-lang.org/std/path/struct.StripPrefixError.html) needs a shorter summary and a link to the method it comes from.
- [ ] [`Component`](https://doc.rust-lang.org/std/path/enum.Component.html) shouldn't say to refer to the module, we should move most of its docs here.
- [ ] [`Prefix`](https://doc.rust-lang.org/std/path/enum.Prefix.html) could use more explanation, and `is_verbatim` needs an example.

### PR DESCRIPTION
**Pull request by chordowl**
Fixes #29368.

This PR contains a pretty significant redistribution of some of the module docs to more appropriate places, as well as general expansion, clarification, and additional examples.

For more details, see the commit descriptions.

r? @steveklabnik

### CODE REVIEW

File: src/libstd/path.rs [hunk 1]
```diff
@@ -143,36 +99,81 @@ use sys::path::{is_sep_byte, is_verbatim_sep, MAIN_SEP_STR, parse_prefix};
...
+/// Windows path prefixes, e.g. `C:` or `\\server\share`.
 ///
-/// Windows uses a variety of path styles, including references to drive
-/// volumes (like `C:`), network shared folders (like `\\server\share`) and
-/// others. In addition, some path prefixes are "verbatim", in which case
-/// `/` is *not* treated as a separator and essentially no normalization is
-/// performed.
+/// Windows uses a variety of path prefix styles, including references to drive
+/// volumes (like `C:`), network shared folders (like `\\server\share`), and
+/// others. In addition, some path prefixes are "verbatim" (i.e. prefixed with
```
Comment (frewsxcv) [inline src/libstd/path.rs]: This parenthesis is never closed

File: src/libstd/path.rs [hunk 2]
```diff
@@ -434,37 +489,49 @@ impl<'a> Hash for PrefixComponent<'a> {
...
+    /// [`Prefix`]: enum.Prefix.html
     #[stable(feature = "rust1", since = "1.0.0")]
     Prefix(
         #[stable(feature = "rust1", since = "1.0.0")] PrefixComponent<'a>
     ),
 
-    /// The root directory component, appears after any prefix and before anything else
+    /// The root directory component, appears after any prefix and before anything else.
+    ///
+    /// It represents a deperator that designates that a path starts from root.
```
Comment (frewsxcv) [inline src/libstd/path.rs]: `deperator` -> `seperator`

**Review by frewsxcv — CHANGES_REQUESTED** [review]
All the changes here are great @lukaramu, awesome work! Just a couple small nits and this should be good to go 👌

**Comment by lukaramu** [comment]
Comments have been addressed, fixed another typo along the way, too! 😅

**Comment by frewsxcv** [comment]
Thanks again!

@bors r+ rollup

**Pull request merged**